src/visitable.cpp

```diff
@@ -72,14 +72,29 @@ bool visitable<T>::has_item_with( const std::function<bool( const item & )> &fil
     } ) == VisitResponse::ABORT;
 }
 
+/** Sums the two terms, being careful to not trigger overflow.
+ * Doesn't handle underflow.
+ *
+ * @param a The first addend.
+ * @param b The second addend.
+ * @return the sum of the addends, but truncated to std::numeric_limits<int>::max().
+ */
+static int sum_no_wrap( int a, int b ) {
```

Comment:
A candidate yes, but it's pointless to do so now as it is only being invoked on a single type.